### DISCUSSION TITLE
PyCharm error

### DISCUSSION THREAD

**Discussion by ForQuestSolutions** [cubewise-code/tm1py#711] [Q&A]
Getting an error:  {NameError}name 'TM1Service' is not defined

Python Code:
with TM1Service(

        address='FORQUEST01',
        port=8004,
        user=r'ForQuest\rstevenson',
        password='pwd',
        namespace='ForQuest',
        ssl=False,
        base_url='http://ForQuest01:8004/api/v1/',
        gateway='',
        decode_b64=True,
        verify = True,
        async_requests_mode = True
        ) as tm1:

    print(tm1.server.get_product_version())

Is the use of the 'r' character correct to handle the backslash \ ?

**Comment by rkvinoth** [comment]
Did you import `TM1Service`?

if not do so,

`from TM1py import TM1Service`

**Comment by ForQuestSolutions** [comment]
I added “from TM1py import TM1Service”.  It’s failing now in the RestService.py script.

Get_Product_Version.py test code.
***@***.***


***@***.***

**Reply by josrobins** [reply]
How do you actually connect to model, i.e. what security model do you use?

It would so be useful to post a complete copy of the code (redacted if necessary), and any error message you are receiving. This will help the community to narrow down the issue and resolve your issue.

**Comment by ForQuestSolutions** [comment]
CAM without SSO

Kind Regards,

Rick Stevenson
Managing Director
908-399-5158

On Apr 7, 2022, at 5:49 PM, Josh.R ***@***.***> wrote:

﻿

How do you actually connect to model, i.e. what security model do you use?

**Reply by josrobins** [reply]
Have you tried dropping the namespace reference from the username? i.e. just 'rstevenson'

**Comment by ForQuestSolutions** [comment]
I have tried that, I try it again.  It works in Postman with the domain prefix.

**Reply by josrobins** [reply]
Dumb question but assume you have encoded your password in base64 given you are using the decode_b64=True?

**Comment by ForQuestSolutions** [comment]
I have the decode_b64 commented out, so password not encoded.

Is the user specified correctly with the r to handle the backspace character?
user=r'ForQuest\rstevenson',

What else can I try?

**Comment by rclapp** [comment]
Can you please provide more information on the exception you are seeing. Copy and Paste the exception if you can.

From: "Josh.R" ***@***.***>
Reply-To: cubewise-code/tm1py ***@***.***>
Date: Thursday, April 7, 2022 at 4:10 PM
To: cubewise-code/tm1py ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [cubewise-code/tm1py] PyCharm error (Discussion #711)


user=r'ForQuest\rstevenson'

**Comment by ForQuestSolutions** [comment]
Let me know if this helps and you can see the 2 images.

***@***.***

***@***.***

**Comment by rclapp** [comment]
No sorry images are not coming through

Sent from my mobile device

On Apr 7, 2022 5:25 PM, Rick ***@***.***> wrote:

Let me know if this helps and you can see the 2 images.

***@***.***

***@***.***

**Comment by ForQuestSolutions** [comment]
![TM1Py 1](https://user-images.githubusercontent.com/59741792/162342191-2c0c44a2-ae97-4524-9330-6c9ee441efa5.jpg)
![TM1Py 2](https://user-images.githubusercontent.com/59741792/162342203-efe144be-0150-4dec-a610-031fab117b97.jpg)

**Reply by cubewise-nswan** [reply]
Thanks, we can see the images now, could you also send an image of the tm1.cfg file

**Reply by ForQuestSolutions** [reply]
![TM1Py 3](https://user-images.githubusercontent.com/59741792/162343263-b34a1ecb-fc47-468f-a28d-ddcb66142ee5.jpg)

**Reply by cubewise-nswan** [reply]
Thanks Rick, could you send the last part of the cfg file

**Reply by ForQuestSolutions** [reply]
![TM1Py 4](https://user-images.githubusercontent.com/59741792/162344774-61b48ebb-4807-45dc-a50c-df8bea88f579.jpg)
![TM1Py 5](https://user-images.githubusercontent.com/59741792/162344779-5611dd0e-d22f-41ee-978c-95484d278eb8.jpg)

**Reply by cubewise-nswan** [reply]
Hi Rick, could you doublecheck the CAM URL setting that you are passing into the authentication. I think it should be the same as the ClientCAMURI setting in the cfg file

**Reply by cubewise-nswan** [reply]
You could also see if the check.py works for you in the samples. (https://code.cubewise.com/tm1py-help-content/check-connectivity-with-tm1 )

**Comment by ForQuestSolutions** [comment]
I posted the images directly, let me know if that worked.

**Comment by ForQuestSolutions** [comment]
Check.py to FOR works.

Kind Regards,

Rick Stevenson
Managing Director
908-399-5158

On Apr 7, 2022, at 9:34 PM, nswan ***@***.***> wrote:

﻿

You could also see if the check.py works for you in the samples. (https://code.cubewise.com/tm1py-help-content/check-connectivity-with-tm1 )

**Reply by cubewise-nswan** [reply]
Hi Rick, this is a good sign. In the TM1Service connection (image you posted earlier), please only include the following parameters/arguments

- address
- port
- user
- password
- ssl
- gateway
- namespace

**Comment by rclapp** [comment]
Just remove base url and you should be okay

Sent from my mobile device

On Apr 7, 2022 8:51 PM, nswan ***@***.***> wrote:

Hi Rick, this is a good sign. In the TM1Service connection (image you posted earlier), please only include the following parameters/arguments

  *   address
  *   port
  *   user
  *   password
  *   ssl
  *   gateway
  *   namespace

-

**Comment by ForQuestSolutions** [comment]
That worked, thanks!
